app/controllers/api/base_controller.rb

```diff
@@ -39,8 +39,10 @@ def verify_with_otp
   end
 
   def verify_mfa_requirement
-    return if @rubygem.mfa_requirement_satisfied_for?(@api_key.user)
-    render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
+    if @rubygem && !@rubygem.mfa_requirement_satisfied_for?(@api_key.user)
+      return render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
+    end
+    render_error_if_user_mfa_requirement_unmet(@api_key.user)
```

Comment:
👍 makes sense to keeping the [user / metadata related checks in one method](https://github.com/rubygems/rubygems.org/pull/3155#discussion_r933363707)!

As a followup, is there a reason why we are delegating  `user.mfa_required_not_yet_enabled?` and `user.mfa_required_weak_level_enabled?` checks to `render_error_if_user_mfa_requirement_unmet(@api_key.user)`? IMO it would be a lot more readable if all the conditionals are in one place.

Reply:
I think the main reason we made it a function is that it the distinction between `gem metadata requiring mfa` and the `user requiring mfa due to downloads` in this function was unclear, so we put it in a function that has "user" in the function name. Also, it may have been a code-climate violation to have so many ifs in one function

Reply:
Ok makes sense, however to make this distinction clear,  I would prefer to have `gem metadata requiring mfa` in another function as well rather than abstracting one check to another function (`user requiring mfa due to downloads`) and not the other. what do you think?

Reply:
We can't extract both methods because we'd still need conditionals within this top level method to generate an early return, otherwise we'd run into a multiple render problem when both conditions are met. 

Now that I look at this with some time and space though, we're calling mfa requirement _on_ the `user`. So i don't think having user in the function name adds much clarity. I went ahead and inlined all the conditionals into one place. Give it another look and let me know what you think @jenshenny @kevinlinxc